component/src/main/java/io/siddhi/extension/io/grpc/source/AbstractGrpcSource.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c)  2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.extension.io.grpc.source;
+
+import io.grpc.ServerBuilder;
+import io.siddhi.core.config.SiddhiAppContext;
+import io.siddhi.core.exception.ConnectionUnavailableException;
+import io.siddhi.core.stream.ServiceDeploymentInfo;
+import io.siddhi.core.stream.input.source.Source;
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.core.util.transport.OptionHolder;
+import io.siddhi.extension.io.grpc.util.GrpcConstants;
+import io.siddhi.extension.io.grpc.util.SourceServerInterceptor;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static io.siddhi.extension.io.grpc.util.GrpcUtils.getServiceName;
+
+/**
+ * This is an abstract class extended by GrpcSource and GrpcServiceSource. This provides most of initialization
+ * implementations common for both sources
+ */
+public abstract class AbstractGrpcSource extends Source {
+    protected SiddhiAppContext siddhiAppContext;
+    protected SourceEventListener sourceEventListener;
+    private String url;
+    private String serviceName;
+    protected boolean isDefaultMode;
+    private int port;
+    protected SourceServerInterceptor serverInterceptor;
+    protected ServerBuilder serverBuilder;
+    protected int serverShutdownWaitingTime;
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
```

Comment:
This should not be null it should expose the service info (need to add for other sources too)